pontoon/sync/formats/ftl.py

```diff
@@ -0,0 +1,148 @@
+            comments=comments or [],
+            fuzzy=False,
+            order=order,
+        )
+
+    def __repr__(self):
+        return '<L20NEntity {key}>'.format(key=self.key.encode('utf-8'))
+
+
+class L20NResource(ParsedResource):
+    def __init__(self, path, locale, source_resource=None):
+        self.path = path
+        self.locale = locale
+        self.entities = {}
+        self.source_resource = source_resource
+        self.order = 0
+
+        # Copy entities from the source_resource if it's available.
+        if source_resource:
+            for key, entity in source_resource.entities.items():
+                self.entities[key] = L20NEntity(
+                    entity.key,
+                    '',
+                    '',
+                    {},
+                    copy.copy(entity.comments),
+                    entity.order
+                )
+
+        with codecs.open(path, 'r', 'utf-8') as resource:
+            self.structure = FTLParser().parseResource(resource.read())
+
+        def get_comment(obj):
+            return [obj['comment']['content']] if obj['comment'] else []
+
+        def parse_entity(obj, section_comment=[]):
+            if obj['value'] and obj['value']['type'] == 'Pattern':
+                self.entities[obj['id']['name']] = L20NEntity(
+                    obj['id']['name'],
+                    obj['value']['source'],
+                    '',
+                    {None: obj['value']['source']},
+                    section_comment + get_comment(obj),
+                    self.order
+                )
+                self.order += 1
+
+        for obj in self.structure[0]['body']:
+            if obj['type'] == 'Entity':
+                parse_entity(obj)
+
+            elif obj['type'] == 'Section':
+                section_comment = get_comment(obj)
+                for obj in obj['body']:
+                    if obj['type'] == 'Entity':
+                        parse_entity(obj, section_comment)
+
+    @property
+    def translations(self):
+        return sorted(self.entities.values(), key=lambda e: e.order)
+
+    def save(self, locale):
+        """
+        Load the source resource, modify it with changes made to this
+        Resource instance, and save it over the locale-specific
+        resource.
+        """
+        if not self.source_resource:
+            raise SyncError('Cannot save l20n resource {0}: No source resource given.'
+                            .format(self.path))
+
+        with codecs.open(self.source_resource.path, 'r', 'utf-8') as resource:
+            structure = FTLParser().parseResource(resource.read())
+
+        def serialize_entity(obj, entities):
+            entity_id = obj['id']['name']
+            translations = self.entities[entity_id].strings
+
+            if obj['value'] and obj['value']['type'] == 'Pattern':
+                if translations:
+                    obj['value']['elements'][0]['value'] = translations[None]
```

Comment:
I don't think I understand that. If you're recreating a Pattern, you need to also recreate it's source (`obj['value']['source']`), and you probably should regenerate the whole `elements` list.